contracts/MechMarketplace.sol

```diff
@@ -0,0 +1,438 @@
+
+    /// @dev MechMarketplace constructor.
+    /// @param _factory Agent mech factory address.
+    /// @param _minResponceTimeout Min response time in sec.
+    /// @param _maxResponceTimeout Max response time in sec.
+    constructor(address _factory, uint256 _minResponceTimeout, uint256 _maxResponceTimeout) {
+        // Check for zero address
+        if (_factory == address(0)) {
+            revert ZeroAddress();
+        }
+
+        // Check for zero values
+        if (_minResponceTimeout == 0 || _maxResponceTimeout == 0) {
+            revert ZeroValue();
+        }
+
+        // Check for sanity values
+        if (_minResponceTimeout > _maxResponceTimeout) {
+            revert Overflow(_minResponceTimeout, _maxResponceTimeout);
+        }
+
+        // responseTimeout limits
+        if (_maxResponceTimeout > type(uint32).max) {
+            revert Overflow(_maxResponceTimeout, type(uint32).max);
+        }
+
+        owner = msg.sender;
+        minResponseTimeout = _minResponceTimeout;
+        maxResponseTimeout = _maxResponceTimeout;
+        factory = _factory;
+
+        // Record chain Id
+        chainId = block.chainid;
+        // Compute domain separator
+        domainSeparator = _computeDomainSeparator();
+    }
+
+    /// @dev Computes domain separator hash.
+    /// @return Hash of the domain separator based on its name, version, chain Id and contract address.
+    function _computeDomainSeparator() internal view returns (bytes32) {
+        return keccak256(
+            abi.encode(
+                DOMAIN_SEPARATOR_TYPE_HASH,
+                keccak256("MechMarketplace"),
+                keccak256(abi.encode(VERSION)),
+                block.chainid,
+                address(this)
+            )
+        );
+    }
+
+    /// @dev Changes contract owner address.
+    /// @param newOwner Address of a new owner.
+    function changeOwner(address newOwner) external virtual {
+        // Check for the ownership
+        if (msg.sender != owner) {
+            revert OwnerOnly(msg.sender, owner);
+        }
+
+        // Check for the zero address
+        if (newOwner == address(0)) {
+            revert ZeroAddress();
+        }
+
+        owner = newOwner;
+        emit OwnerUpdated(newOwner);
+    }
+
+    /// @dev Changes mech factory address.
+    /// @param newFactory New mech factory address.
+    function changeFactory(address newFactory) external {
+        // Check for the contract ownership
+        if (msg.sender != owner) {
+            revert OwnerOnly(msg.sender, owner);
+        }
+
+        // Check for zero address
+        if (newFactory == address(0)) {
+            revert ZeroAddress();
+        }
+
+        factory = newFactory;
+        emit FactoryUpdated(newFactory);
+    }
+
+    /// @dev Sets mech registration status.
+    /// @param mech Mech address.
+    /// @param status True, if registered, false otherwise.
+    function setMechRegistrationStatus(address mech, bool status) external{
```

Comment:
Only registered mechs are able to handle requests.

Reply:
Please add a test to call this via mech not factory, to prove that branch works

Reply:
It's already there